src/Workspaces/Core/Portable/VirtualChars/VirtualChar.cs

```diff
@@ -0,0 +1,64 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis.VirtualChars
+{
+    /// <summary>
+    /// The Regex parser wants to work over an array of characters, however this array of characters
```

Comment:
Maybe the references to the `Regex parser` should be altered, as `VirtualChar` is know independent for the other pull request?